migration_scripts/bulk_load/src/hybrid_ev.clj

```diff
@@ -0,0 +1,587 @@
+          ALTER COLUMN timestamp_ms       SET NOT NULL,
+          ALTER COLUMN bcs                SET NOT NULL,
+          ADD CONSTRAINT %1s_partition_check CHECK (
+              %2$d <= tx_sequence_number
+          AND tx_sequence_number < %3$d
+          )"
+         (events:partition-name n) lo hi)]
+       (jdbc/execute! db)))
+
+(defn events:attach! [db n lo hi]
+  (->> [(format
+         "ALTER TABLE %s
+          ATTACH PARTITION %s FOR VALUES FROM (%d) TO (%d)"
+         +events+ (events:partition-name n) lo hi)]
+       (jdbc/execute! db)))
+
+(defn events:drop-range-check! [db n]
+  (db/with-table! db (events:partition-name n)
+    "ALTER TABLE %1$s DROP CONSTRAINT %1$s_partition_check"))
+
+;; Table: hybrid_ev_emit_pkg ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn ev-emit-pkg:create! [db]
+  (jdbc/with-transaction [tx db]
+    (db/with-table! db +ev-emit-pkg+
+      "CREATE TABLE %s (
+           package                     BYTEA,
+           tx_sequence_number          BIGINT,
+           event_sequence_number       BIGINT,
+           sender                      BYTEA
+       )")
+    (db/disable-autovacuum! tx +ev-emit-pkg+)))
+
+(defn ev-emit-pkg:constrain! [db]
+  (jdbc/with-transaction [tx db]
+    (db/with-table! tx +ev-emit-pkg+
+      "ALTER TABLE %s
+       ADD PRIMARY KEY (package, tx_sequence_number, event_sequence_number),
+       ALTER COLUMN sender SET NOT NULL")))
+
+(defn ev-emit-pkg:index! [db]
+  (db/with-table! db +ev-emit-pkg+
+    "CREATE INDEX %1$s_sender ON %1$s (
+         sender,
+         package,
+         tx_sequence_number,
+         event_sequence_number
+     )"))
+
+;; Table: hybrid_ev_emit_mod ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn ev-emit-mod:create! [db]
+  (jdbc/with-transaction [tx db]
+    (db/with-table! db +ev-emit-mod+
+      "CREATE TABLE %s (
+           package                     BYTEA,
+           module                      TEXT,
+           tx_sequence_number          BIGINT,
+           event_sequence_number       BIGINT,
+           sender                      BYTEA
+       )")
+    (db/disable-autovacuum! tx +ev-emit-mod+)))
+
+(defn ev-emit-mod:constrain! [db]
+  (jdbc/with-transaction [tx db]
+    (db/with-table! tx +ev-emit-mod+
+      "ALTER TABLE %s
+       ADD PRIMARY KEY (
+           package,
+           module,
+           tx_sequence_number,
+           event_sequence_number
+       ),
+       ALTER COLUMN sender SET NOT NULL")))
+
+(defn ev-emit-mod:index! [db]
+  (db/with-table! db +ev-emit-mod+
+    "CREATE INDEX %1$s_sender ON %s (
+         sender,
+         package,
+         module,
+         tx_sequence_number,
+         event_sequence_number
+     )"))
+
+;; Table: hybrid_ev_event_pkg ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn ev-event-pkg:create! [db]
+  (jdbc/with-transaction [tx db]
+    (db/with-table! db +ev-event-pkg+
+      "CREATE TABLE %s (
+           package                     BYTEA,
```

Comment:
Double checking that this is the package of the event struct type itself, not the package in which the event is emitted right?

Reply:
Ah I see the insertion statements below

Reply:
That's right -- there's two tables: `ev-emit-pkg` and `ev-event-pkg`. The former is the package containing the function called in the PTB that generated the event, and the latter (this one) is the package that introduced the event type.